osf/models/schema_response.py

```diff
@@ -1,23 +1,39 @@
 from django.contrib.contenttypes.fields import GenericForeignKey
 from django.contrib.contenttypes.models import ContentType
 from django.db import models
+from osf.models import RegistrationSchemaBlock
 from osf.models.base import BaseModel, ObjectIDMixin
+from osf.models.schema_response_block import SchemaResponseBlock
 from osf.utils.fields import NonNaiveDateTimeField
 
 
 class SchemaResponse(ObjectIDMixin, BaseModel):
+    '''Collects responses for a schema associated with a parent object.
 
+    SchemaResponse manages to creation, surfacing, updating, and approval of
+    "responses" to the questions on a Registration schema (for example).
+
+    Individual answers are stored in SchemaResponseBlocks and aggregated here
+    via the response_blocks M2M relationship.
+
+    SchemaResponseBlocks can be shared across multiple SchemaResponses, but
+    each SchemaResponseBlock links to the SchemaResponse where it originated.
+    These are referenced on the SchemaResponse using the updated_response_blocks manager.
+    This allows SchemaResponses to also serve as a revision history when
+    users submit updates to the schema form on a given parent object.
+    '''
     schema = models.ForeignKey('osf.RegistrationSchema')
     response_blocks = models.ManyToManyField('osf.SchemaResponseBlock')
     initiator = models.ForeignKey('osf.OsfUser', null=False)
     previous_response = models.ForeignKey(
         'osf.SchemaResponse',
         related_name='updated_response',
-        null=True
+        null=True,
+        blank=True
```

Comment:
Why the `blank=True` changes? `blank` is only used in Django forms.

Reply:
I know that some people say that on the Stackoverflow, but this doesn't work with our system. I believe our `dirty_fields` functionality validates based on blankness, so when we run `full_clean` it improperly validates for this and prevents saving. I am not entirely sure this is the best solution, but some special casing is necessary.